### ISSUE TITLE
Consider merging other FITS column and unit code into desiutil.annotate

### ISSUE THREAD

**Issue by weaverba137** [desihub/desiutil#204]
While working on desihub/desispec#2176, we created or encountered other code to manipulate FITS `BinTableHDU` columns descriptions and units in `desispec.io.util.write_bintable` and `desispec.io.fibermap.annotate_fibermap`. These functions have different high-level functionality, but some low-level code is almost certainly in common with `desiutil.annotate.annotate_fits`.

* `desiutil.annotate.annotate_fits` is for updating an existing `BinTableHDU` in an old file, while taking care to not alter the data in other HDUs.
* `desispec.io.util.write_bintable` is for writing or appending a `BinTableHDU`, but the input is not guaranteed itself to be a `BinTableHDU`.
* `desispec.io.fibermap.annotate_fibermap` uses a pre-defined list of columns to manipulate the output of `desispec.io.fibermap.assemble_fibermap` and other FIBERMAP HDUs in the pipeline.

As part of this we may want to consider moving `write_bintable` entirely into `desiutil` since there is nothing really specific to `desispec` about it.

**Comment by moustakas** [comment]
@weaverba137 were all these tasks done in https://github.com/desihub/desiutil/pull/201? If so, can we close this ticket?

**Comment by weaverba137** [comment]
No, and in fact, this issue was created *after* #201 was closed.
